jwst/cube_build/blot_cube_build.py

```diff
@@ -281,19 +231,135 @@ def blot_overlap_quick(self, model, xcenter, ycenter, xstart,
                 index2d = [iy * blot_xsize + ix for iy in index_y[0] for ix in (index_x[0]+xstart)]
                 blot_flux[index2d] = blot_flux[index2d] + weighted_flux
                 blot_weight[index2d] = blot_weight[index2d] + weight_distance
 
+        t1 = time.time()
+        log.debug("Time to blot median image to input model = %.1f s" % (t1-t0,))
         # done mapping blotted x,y (x_cube, y_cube) to detector
         igood = np.where(blot_weight > 0)
         blot_flux[igood] = blot_flux[igood] / blot_weight[igood]
         blot_flux = blot_flux.reshape((blot_ysize, blot_xsize))
         return blot_flux
+
+    # ************************************************************************
+    def blot_images_nirspec(self):
+        """ Core blotting routine for NIRSPEC
+
+        This is the main routine for blotting the NIRSPEC median sky image back to
+        the detector space and creating a blotting image for each input model.
+        This routine was split from the MIRI routine because the blotting for NIRSpec
+        needs to be done slice by slice and an error in the inverse mapping (sky to
+        detector) mapped too many values back to the detector. This routine adds
+        a check and first pulls out the min and max ra and dec values in the slice
+        and only inverts the slice values back to the detector.
+        1. Loop over every data model to be blotted and find ra,dec,wavelength
+           for every pixel in a valid slice on the detector.
+        2. For each slice get the WCS info for the slice. Find the x,y bounding
+           box values and using only the x,y in the slice find the ra,dec, wavelength
+           for the slice. From this information find the min and max ra and dec
+           on the slice. Pull out those values that fall in this range from the
+           median sky. Using this limited set then map back the ra, dec to
+           x,y on detector.
+        3. Blot these inverted x,y values to detector space.
+
+        """
+        blot_models = datamodels.ModelContainer()
+
+        for model in self.input_models:
+            blot_flux = np.zeros(model.shape, dtype=np.float32)
+            blot_weight = np.zeros(model.shape, dtype=np.float32)
+            blot_ysize, blot_xsize = blot_flux.shape
+            blot_flux = np.ndarray.flatten(blot_flux)
+            blot_weight = np.ndarray.flatten(blot_weight)
+            t0 = time.time()
+            blot = model.copy()
+            blot.err = None
+            blot.dq = None
+            # ___________________________________________________________________
+            ysize, xsize = model.data.shape
+            ycenter = np.arange(ysize)
+            xcenter = np.arange(xsize)
+
+            # for NIRSPEC wcs information accessed seperately for each slice
+            nslices = 30
+            log.info('Looping over 30 slices on NIRSPEC detector, this takes a little while')
+            t0 = time.time()
```

Comment:
t0 got set up in line 281 and now it's being set again.

Reply:
removed one of the t0